File: CS/CustomReportClass/CustomReportClass.Module/CustomXAFReport.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp.Reports;
using DevExpress.Xpo;
using DevExpress.XtraReports;
using DevExpress.XtraReports.Serialization;
using DevExpress.Persistent.Base;

namespace CustomReportClass.Module {
    [RootClass]
    public class CustomXafReport : XafReport {
        protected string additionalInfo;
        public string AdditionalInfo {
            get { return additionalInfo; }
            set { additionalInfo = value; }
        }
        protected override void SerializeProperties(XRSerializer serializer) {
            base.SerializeProperties(serializer); 
            serializer.SerializeString("AdditionalInfo", AdditionalInfo);
        }
        protected override void DeserializeProperties(XRSerializer serializer) {
            base.DeserializeProperties(serializer);
            AdditionalInfo = serializer.DeserializeString("AdditionalInfo", string.Empty);
        }
    }
    [System.ComponentModel.DisplayName("Custom Report")]
    [NavigationItem("Reports")]
    public class CustomReportData : ReportData {
        public CustomReportData(Session session) : base(session) { }
        protected override XafReport CreateReport() {
            return new CustomXafReport();
        }
    }
}

File: VB/CustomReportClassVB/CustomReportClass.Module/CustomXAFReport.vb
Imports System.Text
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.Xpo
Imports DevExpress.XtraReports
Imports DevExpress.XtraReports.Serialization
Imports DevExpress.Persistent.Base

Namespace CustomReportClass.Module
    <RootClass> _
    Public Class CustomXafReport
        Inherits XafReport

        Protected _additionalInfo As String
        Public Property AdditionalInfo() As String
            Get
                Return _additionalInfo
            End Get
            Set(ByVal value As String)
                _additionalInfo = value
            End Set
        End Property
        Protected Overrides Sub SerializeProperties(ByVal serializer As XRSerializer)
            MyBase.SerializeProperties(serializer)
            serializer.SerializeString("AdditionalInfo", AdditionalInfo)
        End Sub
        Protected Overrides Sub DeserializeProperties(ByVal serializer As XRSerializer)
            MyBase.DeserializeProperties(serializer)
            AdditionalInfo = serializer.DeserializeString("AdditionalInfo", String.Empty)
        End Sub
    End Class
    <System.ComponentModel.DisplayName("Custom Report"), NavigationItem("Reports")> _
    Public Class CustomReportData
        Inherits ReportData

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Protected Overrides Function CreateReport() As XafReport
            Return New CustomXafReport()
        End Function
    End Class
End Namespace
